include/arch/xtensa/sys_io.h

```diff
@@ -106,6 +106,17 @@ static ALWAYS_INLINE
 	return ret;
 }
 
+static ALWAYS_INLINE
```

Comment:
sys_io.h is meant to hold generic functions for all arch (each needing to implement these then).

So either you generalize sys_set/clear_mask32 in include/sys_io.h or you just create xtensia dedicated ones (as it used to be in fact) but put them in xtensia soc.h 

The later option seems better imo.

Reply:
Makes sense for me.